### DISCUSSION TITLE
Way to implement the /ping and /heathz with different auth method than the usual calls.

### DISCUSSION THREAD

**Discussion by SHAANBHAYA** [mar10/wsgidav#267] [Q&A]
Hi Awesome people who contribute,
Wanted to know if there is any way to configure custom endpoints like /ping or /healthz on the provider.
We have to integrate our webdav server with a load balancer and so need these endpoints.

I understand that we can do this by adding two new DavProvider mappings, but we do not want the domain controller to do auth on these two api calls as they need to be unauthenticated.

We have the config like this

```py
custom_config.update({'server': 'gunicorn', 'provider_mapping': {'/webdav': app}, 'user_mapping': {}, 'verbose': 0,
                          'enable_loggers': [], 'propsmanager': False, 'locksmanager': False,
                          'domaincontroller': inSyncWDDomainController(timeout=900, lh_srv=lh_client),
                          'defaultdigest': False, 'acceptdigest': False, 'acceptbasic': True,
                          'catchall': True, 'middleware_stack': [WsgiDavDirBrowser, CustomLogger, HTTPAuthenticator,
                                                                 ErrorPrinterCustomForWebdavSrv,]})
```

Is there a way to achieve this ?

**Comment by mar10** [comment]
Another way could be to write a new middleware that handles those URLs and add it to [the stack](https://wsgidav.readthedocs.io/en/latest/reference_guide_architecture.html#wsgi-application-stack).

The [CORS middleware](https://github.com/mar10/wsgidav/blob/master/wsgidav/mw/cors.py) may serve as an example.
